src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.cs

```diff
@@ -121,4 +129,19 @@ private static string GetContextBaseUri(HttpRequest request)
         // it has to end with a trailing slash
         return result.EndsWith('/') ? result : result += "/";
     }
+
+    private static IReadOnlyDictionary<string, object> GetRouteDataParameters(HttpContext httpContext)
+    {
+        var parameters = new Dictionary<string, object>();
+
+        var routeValueDictionary = httpContext.GetRouteData().Values;
+        foreach (var kvp in routeValueDictionary)
+        {
+            if (kvp.Value != null)
+            {
+                parameters[kvp.Key] = kvp.Value;
+            }
+        }
+        return parameters;
```

Comment:
Why are we filtering for null here?

In fact, why do we even need to make a copy of the dictionary at all?

Reply:
It is because of `RouteValueDictionay : IReadOnlyDictionary<string, object?>` and `RouteData.RouteValues` is `IReadOnlyDictionary<string, object>`. There is [CS8620](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings?f1url=%3FappId%3Droslyn%26k%3Dk(CS8620)). Another alternative is to suppress it. What do you think?

Reply:
@surayya-MS I think we should update the nullability annotation in `RouteData.RouteValues` to match the one in `RouteValueDictionary`.

I believe it is considered a "breaking change", but that we have always sided on "correctness" in this case.